docs/managedCodeGen.md

```diff
@@ -0,0 +1,29 @@
+Doing it at build time should result in developers seeing all errors up front, rather than only seeing errors for the code being tested. And the runtime cost is not incurred every single time.
+
+## How it works
+This change adds a new targets file that is imported from the [CSharpLib.targets](https://github.com/microsoft/react-native-windows/blob/master/vnext/PropertySheets/External/Microsoft.ReactNative.Uwp.CppLib.targets) and [CSharpApp.targets](https://github.com/microsoft/react-native-windows/blob/master/vnext/PropertySheets/External/Microsoft.ReactNative.Uwp.CppApp.targets). That will enable the codegen for managed projects.
+
+### In source mode
+The targets file hooks up an msbuild target that will ensure the tool is deployed and runnable. The codegen tool is defined in the Microsoft.ReactNative.Managed.CodeGen project. It will build and deploy the tool in the current version of your project (except for ARM and Arm64 it will use x86 and x64). It will then collect all the C# source files, assembly references and defines from the app (or library) and invoke the tool to generate a C# file. This is very similar to how Xaml codegen works.
+The C# file will contain a class called that implements
+
+### In nuget mode (Future)
+The tool will be shipped as a binary and the steps won't have to build or deploy the tool. We will also ship one version of the tool, rather than all qualifiers. 
+When building the apps that are checked in to our repo for testing, will remain running off the source versoin.
+
+## Turning on/off
+Currently the feature is behind an msbuild property `$(ReactNativeCodeGenEnabled)`.
+The default at the moment is false, it is only turned  on for a single project for now which is the [SampleLibraryCS.csproj](https://github.com/microsoft/react-native-windows/blob/master/packages/microsoft-reactnative-sampleapps/windows/SampleLibraryCS/SampleLibraryCS.csproj) project, to prove it is stable. We will slowly enable it for all projects in the repo and then make it the default.
+
+## MsBuild/Nuget Complications
+MsBuild and Nuget spent a long time fighting me in mixing a NetCoreApp2.2 executable and WinRT apps in the same solution and the same build. ProjectReferences cannot be used so I had to use the `<MsBuild>` task directly in the targets and it was tricky making it build from both the customer apps as well as our main build solution and unittest (Microsoft.ReactNative.sln). In the end there are a few hacks in place to make this work.
+The hookups in msbuild do not use the latest BeforeTargets/AfterTargets feature to schedule the main work. I opted to follow the same pattern that the Xaml Codegen uses, so the React CodeGen runs after all the other codegen tools (like resgen, xamlcompile etc) to ensure people familiar with those extensions know how to work with the React Managed CodeGen.
+
+## Future improvements
+ * Add support for `Task<T>` async methods.
+ * Ship it as a nuget package
```

Comment:
are there issues filed for these? can you reference them here?

Reply:
Yes, will add nrs